p2p/discovery/addrbook.go

```diff
@@ -90,8 +91,8 @@ type addrBook struct {
 	addrNew   [newBucketCount]map[node.ID]*KnownAddress
 	addrTried [triedBucketCount]map[node.ID]*KnownAddress
 
-	//todo: lock localNode for updates
-	localAddress *node.NodeInfo
+	//todo: lock local for updates
+	localAddresses []*node.NodeInfo
```

Comment:
why array?

Reply:
I figured out an array would be useful because eventually you might have a new public ip, also you add your key at the beginning while you don't know your public ip, so I thought for simplicity we could use a completely different entry for the ip/key. 
though we're not adding ips yet since we don't really know our public ip atm.

Reply:
also after a restart you might have a new public key, but since we persist the address book we'll save the old one as well to prevent you trying to connect yourself even if you can't tell who is the ip.